ble/BLEProtocol.h

```diff
@@ -40,10 +47,82 @@ namespace BLEProtocol {
             RANDOM_PRIVATE_NON_RESOLVABLE
         };
     };
-    typedef AddressType::Type AddressType_t; /**< Alias for AddressType::Type */
+    typedef AddressType::Type AddressType_t;   /**< Alias for AddressType::Type */
 
-    static const size_t ADDR_LEN = 6;        /**< Length (in octets) of the BLE MAC address. */
-    typedef uint8_t Address_t[ADDR_LEN];     /**< 48-bit address, in LSB format. */
+    static const size_t ADDR_LEN = 6;         /**< Length (in octets) of the BLE MAC address. */
+    typedef uint8_t AddressBytes_t[ADDR_LEN]; /**< 48-bit address bytes, in LSB format. */
+
+    /**
+     * BLE address. It contains an address-type (@ref AddressType_t) and bytes (@ref AddressBytes_t).
+     */
+    struct Address_t {
+        AddressType_t  type;    /**< @ref AddressType_t */
+        AddressBytes_t address; /**< @ref AddressBytes_t */
+
+        Address_t(AddressType_t typeIn, const AddressBytes_t addressIn) : type(typeIn) {
+            std::copy(addressIn, addressIn + ADDR_LEN, address);
+        }
+    };
+
+    static const size_t SECURITY_KEY_LEN = 16; /**@brief GAP Security Key Length. */
+    typedef uint8_t Irk_t[SECURITY_KEY_LEN];   /**@brief Identity Resolving Key. */
+
+    /**
+     * Whitelisting is an important feature available in BLE. White-lists allow
+     * hosts to filter devices when advertising, scanning, and establishing
+     * connections on both sides. White lists are simply arrays of Bluetooth
+     * device addresses that are populated by the host (i.e. the processor where
+     * the application is running) and stored and used in the controller (i.e.
+     * Bluetooth controller).
+     *
+     * A device scanning or initiating a connection can use a white list to
+     * limit the number of devices that will be detected or with which it can
+     * connect, and the advertising device can use a white list to specify which
+     * peers it will accept an incoming connection from. This is especially
+     * useful when searching for a few known devices in a fog of advertising
+     * packets in busy locations.
+     *
+     * Any advertising packets (in the case of a scanner) or connection request
+     * packets (in the case of an advertiser) received from devices whose
+     * Bluetooth Address is not present in the white list will simply be
+     * dropped.
+     *
+     * @note A whitelist has a fixed size. There will be some upper limit
+     *     imposed by the underlying BLE stack.
+     *
+     * @note It is not possible to change the contents of the whitelist while it
```

Comment:
Perhaps I am misunderstanding this limitation, but I am under the impression that essentially a device would like to update its white list just after it connects to a device. But if is advertising other services then it does not look like it can.